File: src/main/java/io/github/malczuuu/iemu/domain/LightModule.java
/*
 * Copyright (c) 2025 Damian Malczewski
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * SPDX-License-Identifier: MIT
 */
package io.github.malczuuu.iemu.domain;

import java.util.function.Consumer;

public interface LightModule {

  void setOn(boolean on);

  void setDimmer(int dimmer);

  void setOnTime(long onTime);

  boolean isOn();

  int getDimmer();

  long getOnTime();

  void shutdown();

  void subscribeOnStateChange(Consumer<Boolean> consumer);

  void subscribeOnDimmerChange(Consumer<Integer> consumer);

  void subscribeOnTimeCounterChange(Consumer<Long> consumer);
}

File: src/main/kotlin/io/github/malczuuu/iemu/domain/LightModule.kt
package io.github.malczuuu.iemu.domain

interface LightModule {

  var isOn: Boolean

  var dimmer: Int

  var onTime: Long

  fun subscribeOnStateChange(consumer: (Boolean) -> Unit): Boolean

  fun subscribeOnDimmerChange(consumer: (Int) -> Unit): Boolean

  fun subscribeOnTimeCounterChange(consumer: (Long) -> Unit): Boolean

  fun shutdown()
}
